Ginger/GingerCoreNET/ActionsLib/General/ActScreenShot.cs

```diff
@@ -95,84 +91,100 @@ public string SaveToFileName
+            if (!string.IsNullOrEmpty(fileExtension) && fileExtension != DEFAULT_IMAGE_EXTENSION)
+            {
+                throw new InvalidOperationException($"Unsupported file extension '{fileExtension}'. Only {DEFAULT_IMAGE_EXTENSION} is allowed.");
+            }
 
-            string DirectoryPath = SaveToFileName;
+            if (!ScreenShots.Any())
+            {
+                throw new InvalidOperationException("No screenshots were captured for saving.");
+            }
+        }
+
+        private string ManageFilePath()
+        {
+            string directoryPath = Path.GetDirectoryName(SaveToFileName);
 
-            if (DirectoryPath.StartsWith(@"~\"))
+            if (directoryPath.StartsWith(TILDE_PATH_PREFIX))
             {
-                DirectoryPath = Path.Combine(SolutionFolder, DirectoryPath.Replace(@"~\", string.Empty));
+                directoryPath = Path.Combine(SolutionFolder, directoryPath.Replace(TILDE_PATH_PREFIX, string.Empty));
             }
 
-            if (!Directory.Exists(DirectoryPath))
+            if (!Directory.Exists(directoryPath))
             {
                 try
                 {
-                    Directory.CreateDirectory(DirectoryPath);
+                    Directory.CreateDirectory(directoryPath);
                 }
-                catch (Exception ex)
+                catch
                 {
-                    this.Status = Amdocs.Ginger.CoreNET.Execution.eRunStatus.Failed;
-                    Error = string.Concat("Invalid Folder Path. :", DirectoryPath);
-                    throw ex;
+                    Status = Amdocs.Ginger.CoreNET.Execution.eRunStatus.Failed;
+                    throw new InvalidOperationException($"The provided folder path is invalid: {directoryPath}");
                 }
-
             }
 
-            String FileName = this.Description;
-            String timeStamp = DateTime.Now.ToString("ddMMyyyyHHmmss");
-
-            FileName += timeStamp;
-
+            return directoryPath;
+        }
+        private void SaveImages(string directoryPath)
+        {
+            string fileName = Path.GetFileNameWithoutExtension(SaveToFileName);
+            string fileExtension = Path.GetExtension(SaveToFileName);
 
-            if (ScreenShots.Count == 0)
+            if (string.IsNullOrEmpty(fileExtension))
             {
-                Error = "Failed to capture screen shots to save";
-                return;
+                fileName = $"{Description}{DateTime.Now:ddMMyyyyHHmmss}";
+                fileExtension = DEFAULT_IMAGE_EXTENSION;
             }
 
-            //TODO: make it simple file copy + change var to folder to save not file name
-            ObservableList<Bitmap> Bitmp = new ObservableList<Bitmap>();
-            foreach (String path in ScreenShots)
+            Dictionary<string, object> outputFilePaths = new();
+            for (int i = 0; i < ScreenShots.Count; i++)
```

Comment:
This is already taken care of one lines 119-122:

```
if (!ScreenShots.Any())
{
    Error = "No screenshots were captured for saving.";
    return;
}
```

Since `ValidateInput()` runs before the `SaveImages()` we're guaranteed to have some screenshots at this point.

```
ValidateInput();

string directoryPath = ManageFilePath();
SaveImages(directoryPath);
```

Reply:
Also @Maheshkale447, previously `ObservableList` was used here which inherits from `ObservableCollection`, however, I didn't see a need to use that DS as there didn't appear to be any UI binding or reason to notify any listeners if the list changes so I opted for a simple `Dictionary`. Was this a good choice, am I missing something?

Reply:
Its fine